src/main/java/io/tiledb/spark/TileDBDataReaderPartitionScan.java

```diff
@@ -344,6 +344,27 @@ private boolean initQuery() throws TileDBError {
         dimIndex++;
       }
 
+      // if array is dense, creating a subarray is necessary
+      if (!arraySchema.isSparse()) {
+        Datatype dimType = domain.getDimension(0).getType(); // first dim is enough
+        NativeArray subarray = new NativeArray(ctx, 2 * ((int) domain.getNDim()), dimType);
+        int subarrayIndex = 0;
+        for (int d = 0; d < domain.getNDim(); d++) {
+          subarray.setItem(
+              subarrayIndex,
+              dimensionRanges.get(d).getFirst() != null
+                  ? dimensionRanges.get(d).getFirst()
+                  : Integer.MIN_VALUE); // required by dense arrays
```

Comment:
This (and below) doesn't seem right -- the subarray is going to be enormous if the ranges are not set. Should we just error out of am I missing something here?